### PR TITLE
feat(lock): Implement the Lock API on HTTP

### PR DESCRIPTION
**Pull request by XavierGeerinck**
# Description

Implemented the Lock API

Note: Tests are still WIP and most likely we have a small breaking change on the gRPC part as we named it `tryLock` before which is not the way JS implements it or the spec names it

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #414 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [x] Created/updated tests
- [x] Extended the documentation

### CODE REVIEW

**Comment by XavierGeerinck** [comment]
Note: lock is alpha, so we decided not to do a breaking release but a major one

**Comment by shubham1172** [comment]
ping @XavierGeerinck

File: pnpm-lock.yaml
Comment (shubham1172) [inline pnpm-lock.yaml:1]: What is this file? Is this required?

**Review by shubham1172 — COMMENTED** [review]
LGTM, one minor comment.

**Comment by XavierGeerinck** [comment]
Done!

**Pull request merged**